src/script/script.h

```diff
@@ -328,8 +329,7 @@ class CScriptNum
             return std::vector<unsigned char>();
 
         std::vector<unsigned char> result;
-        const bool neg = value < 0;
-        uint64_t absvalue = neg ? -value : value;
+        uint64_t absvalue = static_cast<uint64_t>(std::llabs(value));
```

Comment:
I think you want to use `std::abs`, which since C++11 is overloaded to work for all signed integer types as input (avoiding the assumption that long long is enough for 64 bits, and also avoiding promoting values to higher than 64 bits in case long long is even bigger).

Reply:
OK thanks I'm testing this. I thought that `llabs` showed more clearly the intend but using `abs` might be better if you want to switch to a smaller type later indeed.

Reply:
Oh no, this won't ever switch to a smaller type.

I'm just commenting this because hardcoding the assumption that long long = 64 bits seems bad (if there were an abs64bit or so that would seem ideal, but there isn't). Thankfully, thanks to std::abs being overloaded for all integer types, there is no need for such an assumption.

Reply:
Hah indeed yes, this is better that way then.